File: common/src/main/java/juuxel/adorn/block/property/FrontConnection.java
package juuxel.adorn.block.property;

import net.minecraft.util.StringRepresentable;

/**
 * Note: sofas have left and right flipped
 */
public enum FrontConnection implements StringRepresentable {
    /**
     * No connection to the block in front.
     */
    NONE("none"),

    /**
     * The block will be connected to the block in front on its left side (looking from the connecting block).
     */
    LEFT("left"),

    /**
     * The block will be connected to the block in front on its right side (looking from the connecting block).
     */
    RIGHT("right");

    private final String id;

    FrontConnection(String id) {
        this.id = id;
    }

    @Override
    public String getSerializedName() {
        return id;
    }
}

File: src/main/kotlin/juuxel/adorn/util/FrontConnection.kt
package juuxel.adorn.util

import net.minecraft.util.SnakeCaseIdentifiable

/**
 * Note: sofas have left and right flipped
 */
enum class FrontConnection(private val value: String) : SnakeCaseIdentifiable {
    /**
     * No connection to the block in front.
     */
    None("none"),

    /**
     * The block will be connected to the block in front on its left side (looking from the connecting block).
     */
    Left("left"),

    /**
     * The block will be connected to the block in front on its right side (looking from the connecting block).
     */
    Right("right");

    override fun toSnakeCase() = value
}
